pio/ws2812/ws2812_parallel.c

```diff
@@ -16,7 +16,9 @@
 #include "ws2812.pio.h"
 
 #define FRAC_BITS 4
-#define PIN_TX 0
+#define PIN_TX 2
+// make sure that this number is less or equal to the MAX_LENGTH defined below
+#define NUM_PIXELS 64
```

Comment:
Perhaps it's worth moving the MAX_LENGTH define up to here, and then you could add a compile-time `#error` if `NUM_PIXELS > MAX_LENGTH` ?

Reply:
We could also get rid of MAX_LENGTH altogether and just use NUM_PIXELS. But I do not understand the code good enough if there are some deeper reasons for defining a MAX_LENGTH and use a different number of LEDs.

Reply:
_Maybe_ the intention was that `n` could be different for different strings (just like `current_string_4colour` is different for the different strings), with `MAX_LENGTH` being the longest string allowed? :shrug:  There's so much going on in this code that I also can't quite tell what it is doing (without spending more time looking at it than I have free right now).

Reply:
Even if it was the intention, it does not work that way. I removed n and MAX_LENGTH and use NUM_PIXELS instead. You can still use two strips with different length, just define the number of the longer strip.